src/units/abilities.cpp

```diff
@@ -1229,6 +1229,40 @@ namespace { // Helpers for attack_type::special_active()
 
 }//anonymous namespace
 
+unit_ability_list attack_type::get_specials_and_abilities_with_priority(const std::string& special) const
```

Comment:
I suggest putting this directly beneath `get_specials_and_abilities`, as it continues on from that code.

Reply:
then all weapon specials could use that.

Reply:
i refuse to doing that because mone specials whreturn string value must be concerned by that.

Reply:
excuse me, i don't understand that what uoou said. I will  relocate function

Reply:
function relocated